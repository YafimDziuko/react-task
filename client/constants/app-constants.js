export const SEARCH = {
    genres: 'genres',
    title: 'title'
};

export const SORT = {
    rating: "vote_average",
    releaseDate: "release_date"
}

export const ACTIONS = {
    RETURN_TO_MAIN_PAGE: "RETURN_TO_MAIN_PAGE",
    UPDATE_INPUT_VALUE: "UPDATE_INPUT_VALUE",
    RESET_INPUT_VALUE: "RESET_INPUT_VALUE",
    SEARCH: "SEARCH",
    SWITCH_SEARCH: "SWITCH_SEARCH",
    SWITCH_SORT: "SWITCH_SORT",
    FULL_FILM_LOAD: "FULL_FILM_LOAD",
    PAGINATION_HANDLER: "PAGINATION_HANDLER"
}

export const ITEM_COUNT_PER_PAGE = 12;
export const PAGE_RANGE_DISPLAYED = 5;

export const INITIAL_STATE = {
    searchInfo: {},
    results: {
        data:[]
    },
    fullItem: {
        isActive: false
    },
    inputValue: "",
    sort: "releaseDate",
    search: 'title',
    resultsCount: -1
}