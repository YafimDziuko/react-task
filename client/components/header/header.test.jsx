import React from 'react';
import Header from './header';
import configureStore from 'redux-mock-store';
import { fetch } from 'isomorphic-fetch';

import FullFilmItem from '../full-film-item/full-film-item';
import { INITIAL_STATE } from '../../constants/app-constants';

const mockStore = configureStore();
let store;

describe('<Header/>', function() {
    beforeEach(() => {
        store = mockStore(INITIAL_STATE);
    });

    it('should render Header and match snapshot', function() {
        const wrapper = shallow(<Header store={store}/>).dive();

        expect(wrapper).toMatchSnapshot();
    });

    it('should render Header with Full Film Item', function() {
        store = mockStore({...INITIAL_STATE, fullItem: {isActive: true}});
        const wrapper = shallow(<Header store={store}/>).dive();

        expect(wrapper).toMatchSnapshot();
    });
});

describe('Click events must work as well', function() {  
    beforeEach(() => {
        store = mockStore(INITIAL_STATE);
    });

    it('Genre button', () => {
        const wrapper = shallow(<Header store={store}/>).dive();
        const instance = wrapper.instance();
        wrapper.update();
        const buttons = wrapper.find('button');

        buttons.at(0).simulate('click');
        expect(store.getActions()).toEqual([ { type: 'SWITCH_SEARCH', search: 'genres' } ]);
    });

    it('Title Button', () => {
        const wrapper = shallow(<Header store={store}/>).dive();
        const instance = wrapper.instance();
        wrapper.update();
        const buttons = wrapper.find('button');

        buttons.at(1).simulate('click');
        expect(store.getActions()).toEqual([ { type: 'SWITCH_SEARCH', search: 'title' } ]);
    });

    // it('Search Button', () => {
    //     const getMovie = jest.fn((() => {
    //         console.log('hello');
    //     }));
    //     const wrapper = shallow(<Header store={store}/>).dive();
    //     console.log(wrapper.instance().props.store.getActions());
    //     const buttons = wrapper.find('button');

    //     buttons.at(2).simulate('click');
    //     expect(getMovie).toHaveBeenCalled();
    // });

    it('should return to main page by clicking on Return Butoon', () => {
        store = mockStore({...INITIAL_STATE, fullItem: { isActive: true }});
        const wrapper = shallow(<Header store={store}/>).dive();
        const buttons = wrapper.find('button');

        buttons.at(0).simulate('click');
        expect(store.getActions()).toEqual(  [ { type: 'RETURN_TO_MAIN_PAGE' } ]);
    });

    
    it('should reset input value', () => {
        store = mockStore({...INITIAL_STATE, inputValue: 'transformers'});
        const wrapper = shallow(<Header store={store}/>).dive();
        const instance = wrapper.instance();
        wrapper.update();
        const images = wrapper.find('img');

        images.at(0).simulate('click');
        expect(store.getActions()).toEqual([ { type: 'RESET_INPUT_VALUE'} ]);
    });

    // it('should update input change', () => {
    //     const wrapper = shallow(<Header store={store}/>).dive();
    //     const instance = wrapper.instance();
    //     jest.spyOn(instance, 'updateInputValue');
    //     jest.spyOn(instance, 'getMovie');
    //     wrapper.update();

    //     const inputs = wrapper.find('input');
    //     inputs.at(0).simulate('change', {target: {value: 'matched'} });
    //     expect(instance.updateInputValue).toHaveBeenCalled();

    //     inputs.at(0).simulate('keyPress', {keyCode: 13});
        
    //     expect(store.getActions()[0]).toEqual({ type: 'UPDATE_INPUT_VALUE', value: 'matched' });
    // });
  });
