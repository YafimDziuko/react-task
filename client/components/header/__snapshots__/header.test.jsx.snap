// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> should render Header and match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    filmItem={false}
    fullItem={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="return-div"
        >
          <Nextflixroulette />
        </div>,
        <div>
          <p
            className="find-movie"
          >
            FIND YOUR MOVIE
          </p>
          <div
            className="search-wrapper"
          >
            <input
              className="search-bar"
              onChange={[Function]}
              onClick={[Function]}
              placeholder="Let's find your movie"
              value=""
            />
            <img
              src="test-file-stub"
            />
          </div>
          <div
            className="button-group cl-white"
          >
            <div>
              <p
                className="search-by"
              >
                SEARCH BY
              </p>
              <button
                className="btn genre-button bg-grey cl-white"
                onClick={[Function]}
              >
                GENRE
              </button>
              <button
                className="btn title-button bg-red cl-white"
                onClick={[Function]}
              >
                TITLE
              </button>
            </div>
            <button
              className="btn search-button bg-red cl-white"
              onClick={[Function]}
            >
              SEARCH
            </button>
          </div>
        </div>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Nextflixroulette />,
            false,
          ],
          "className": "return-div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          false,
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="find-movie"
            >
              FIND YOUR MOVIE
            </p>,
            <div
              className="search-wrapper"
            >
              <input
                className="search-bar"
                onChange={[Function]}
                onClick={[Function]}
                placeholder="Let's find your movie"
                value=""
              />
              <img
                src="test-file-stub"
              />
            </div>,
            <div
              className="button-group cl-white"
            >
              <div>
                <p
                  className="search-by"
                >
                  SEARCH BY
                </p>
                <button
                  className="btn genre-button bg-grey cl-white"
                  onClick={[Function]}
                >
                  GENRE
                </button>
                <button
                  className="btn title-button bg-red cl-white"
                  onClick={[Function]}
                >
                  TITLE
                </button>
              </div>
              <button
                className="btn search-button bg-red cl-white"
                onClick={[Function]}
              >
                SEARCH
              </button>
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "FIND YOUR MOVIE",
              "className": "find-movie",
            },
            "ref": null,
            "rendered": "FIND YOUR MOVIE",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  className="search-bar"
                  onChange={[Function]}
                  onClick={[Function]}
                  placeholder="Let's find your movie"
                  value=""
                />,
                <img
                  src="test-file-stub"
                />,
              ],
              "className": "search-wrapper",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "search-bar",
                  "onChange": [Function],
                  "onClick": [Function],
                  "placeholder": "Let's find your movie",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "src": "test-file-stub",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div>
                  <p
                    className="search-by"
                  >
                    SEARCH BY
                  </p>
                  <button
                    className="btn genre-button bg-grey cl-white"
                    onClick={[Function]}
                  >
                    GENRE
                  </button>
                  <button
                    className="btn title-button bg-red cl-white"
                    onClick={[Function]}
                  >
                    TITLE
                  </button>
                </div>,
                <button
                  className="btn search-button bg-red cl-white"
                  onClick={[Function]}
                >
                  SEARCH
                </button>,
              ],
              "className": "button-group cl-white",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p
                      className="search-by"
                    >
                      SEARCH BY
                    </p>,
                    <button
                      className="btn genre-button bg-grey cl-white"
                      onClick={[Function]}
                    >
                      GENRE
                    </button>,
                    <button
                      className="btn title-button bg-red cl-white"
                      onClick={[Function]}
                    >
                      TITLE
                    </button>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "SEARCH BY",
                      "className": "search-by",
                    },
                    "ref": null,
                    "rendered": "SEARCH BY",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "GENRE",
                      "className": "btn genre-button bg-grey cl-white",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "GENRE",
                    "type": "button",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "TITLE",
                      "className": "btn title-button bg-red cl-white",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "TITLE",
                    "type": "button",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "SEARCH",
                  "className": "btn search-button bg-red cl-white",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "SEARCH",
                "type": "button",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="return-div"
          >
            <Nextflixroulette />
          </div>,
          <div>
            <p
              className="find-movie"
            >
              FIND YOUR MOVIE
            </p>
            <div
              className="search-wrapper"
            >
              <input
                className="search-bar"
                onChange={[Function]}
                onClick={[Function]}
                placeholder="Let's find your movie"
                value=""
              />
              <img
                src="test-file-stub"
              />
            </div>
            <div
              className="button-group cl-white"
            >
              <div>
                <p
                  className="search-by"
                >
                  SEARCH BY
                </p>
                <button
                  className="btn genre-button bg-grey cl-white"
                  onClick={[Function]}
                >
                  GENRE
                </button>
                <button
                  className="btn title-button bg-red cl-white"
                  onClick={[Function]}
                >
                  TITLE
                </button>
              </div>
              <button
                className="btn search-button bg-red cl-white"
                onClick={[Function]}
              >
                SEARCH
              </button>
            </div>
          </div>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Nextflixroulette />,
              false,
            ],
            "className": "return-div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            false,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="find-movie"
              >
                FIND YOUR MOVIE
              </p>,
              <div
                className="search-wrapper"
              >
                <input
                  className="search-bar"
                  onChange={[Function]}
                  onClick={[Function]}
                  placeholder="Let's find your movie"
                  value=""
                />
                <img
                  src="test-file-stub"
                />
              </div>,
              <div
                className="button-group cl-white"
              >
                <div>
                  <p
                    className="search-by"
                  >
                    SEARCH BY
                  </p>
                  <button
                    className="btn genre-button bg-grey cl-white"
                    onClick={[Function]}
                  >
                    GENRE
                  </button>
                  <button
                    className="btn title-button bg-red cl-white"
                    onClick={[Function]}
                  >
                    TITLE
                  </button>
                </div>
                <button
                  className="btn search-button bg-red cl-white"
                  onClick={[Function]}
                >
                  SEARCH
                </button>
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "FIND YOUR MOVIE",
                "className": "find-movie",
              },
              "ref": null,
              "rendered": "FIND YOUR MOVIE",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    className="search-bar"
                    onChange={[Function]}
                    onClick={[Function]}
                    placeholder="Let's find your movie"
                    value=""
                  />,
                  <img
                    src="test-file-stub"
                  />,
                ],
                "className": "search-wrapper",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "search-bar",
                    "onChange": [Function],
                    "onClick": [Function],
                    "placeholder": "Let's find your movie",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "src": "test-file-stub",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div>
                    <p
                      className="search-by"
                    >
                      SEARCH BY
                    </p>
                    <button
                      className="btn genre-button bg-grey cl-white"
                      onClick={[Function]}
                    >
                      GENRE
                    </button>
                    <button
                      className="btn title-button bg-red cl-white"
                      onClick={[Function]}
                    >
                      TITLE
                    </button>
                  </div>,
                  <button
                    className="btn search-button bg-red cl-white"
                    onClick={[Function]}
                  >
                    SEARCH
                  </button>,
                ],
                "className": "button-group cl-white",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <p
                        className="search-by"
                      >
                        SEARCH BY
                      </p>,
                      <button
                        className="btn genre-button bg-grey cl-white"
                        onClick={[Function]}
                      >
                        GENRE
                      </button>,
                      <button
                        className="btn title-button bg-red cl-white"
                        onClick={[Function]}
                      >
                        TITLE
                      </button>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "SEARCH BY",
                        "className": "search-by",
                      },
                      "ref": null,
                      "rendered": "SEARCH BY",
                      "type": "p",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "GENRE",
                        "className": "btn genre-button bg-grey cl-white",
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": "GENRE",
                      "type": "button",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "TITLE",
                        "className": "btn title-button bg-red cl-white",
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": "TITLE",
                      "type": "button",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "SEARCH",
                    "className": "btn search-button bg-red cl-white",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "SEARCH",
                  "type": "button",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<Header/> should render Header with Full Film Item 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    filmItem={false}
    fullItem={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="return-div"
        >
          <Nextflixroulette />
          <button
            className="btn return-button cl-red bg-white"
            onClick={[Function]}
          >
            SEARCH
          </button>
        </div>,
        <FullFilmItem
          additional="Oscar movie"
          description="A film crew is a group of people hired by a film company, employed during the phase, for the purpose of producing a film or motion picture. Crew is distinguished from cast, who are the actors who appear in front of the camera or provide voices for characters in the film."
          duration={124}
          genre="horror"
          posterUrl="http://bipbap.ru/wp-content/uploads/2017/05/VOLKI-krasivye-i-ochen-umnye-zhivotnye.jpg"
          rating={4.7}
          title="Wolf wood"
          year={1992}
        />,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Nextflixroulette />,
            <button
              className="btn return-button cl-red bg-white"
              onClick={[Function]}
            >
              SEARCH
            </button>,
          ],
          "className": "return-div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "SEARCH",
              "className": "btn return-button cl-red bg-white",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "SEARCH",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "additional": "Oscar movie",
          "description": "A film crew is a group of people hired by a film company, employed during the phase, for the purpose of producing a film or motion picture. Crew is distinguished from cast, who are the actors who appear in front of the camera or provide voices for characters in the film.",
          "duration": 124,
          "genre": "horror",
          "posterUrl": "http://bipbap.ru/wp-content/uploads/2017/05/VOLKI-krasivye-i-ochen-umnye-zhivotnye.jpg",
          "rating": 4.7,
          "title": "Wolf wood",
          "year": 1992,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="return-div"
          >
            <Nextflixroulette />
            <button
              className="btn return-button cl-red bg-white"
              onClick={[Function]}
            >
              SEARCH
            </button>
          </div>,
          <FullFilmItem
            additional="Oscar movie"
            description="A film crew is a group of people hired by a film company, employed during the phase, for the purpose of producing a film or motion picture. Crew is distinguished from cast, who are the actors who appear in front of the camera or provide voices for characters in the film."
            duration={124}
            genre="horror"
            posterUrl="http://bipbap.ru/wp-content/uploads/2017/05/VOLKI-krasivye-i-ochen-umnye-zhivotnye.jpg"
            rating={4.7}
            title="Wolf wood"
            year={1992}
          />,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Nextflixroulette />,
              <button
                className="btn return-button cl-red bg-white"
                onClick={[Function]}
              >
                SEARCH
              </button>,
            ],
            "className": "return-div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "SEARCH",
                "className": "btn return-button cl-red bg-white",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "SEARCH",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "additional": "Oscar movie",
            "description": "A film crew is a group of people hired by a film company, employed during the phase, for the purpose of producing a film or motion picture. Crew is distinguished from cast, who are the actors who appear in front of the camera or provide voices for characters in the film.",
            "duration": 124,
            "genre": "horror",
            "posterUrl": "http://bipbap.ru/wp-content/uploads/2017/05/VOLKI-krasivye-i-ochen-umnye-zhivotnye.jpg",
            "rating": 4.7,
            "title": "Wolf wood",
            "year": 1992,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
